src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedVariantContextUtils.java

```diff
@@ -402,4 +405,143 @@ public static boolean isComplexIndel(final Allele ref, final Allele allele) {
         }
     }
+
+    /**
+     * Given a set of alleles (reference and alternate), choose the allele that is the best match for the given read (and offset)
+     *
+     * @param pileupElement read and offset.  Never {@code null}
+     * @param referenceAllele Reference allele.  Never {@code null}
+     * @param altAlleles List of candidate alternate alleles.  Never {@code null}
+     * @param minBaseQualityCutoff minimum base quality for the bases that match the allele in order to be counted.
+     *                             Must be positive or zero.  If you do not want any filtering, specify 0.
+     * @return The allele (reference or from the altAlleles) that matches.  {@code null} if none are a match or the base qualities
+     *      corresponding to the allele don't all exceed the minimum.
+     */
+    public static Allele chooseAlleleForRead(final PileupElement pileupElement, final Allele referenceAllele, final List<Allele> altAlleles, int minBaseQualityCutoff) {
+        Utils.nonNull(pileupElement);
+        Utils.nonNull(referenceAllele);
+        Utils.nonNull(altAlleles);
+        ParamUtils.isPositiveOrZero(minBaseQualityCutoff, "Minimum base quality must be positive or zero.");
+
+        final boolean isRef = referenceAllele.basesMatch(getBasesForAlleleInRead(pileupElement, referenceAllele))
+                && !pileupElement.isBeforeDeletionStart() && !pileupElement.isBeforeInsertion();
+
+        Allele pileupAllele = null;
+        if (!isRef) {
+
+            for (Allele altAllele : altAlleles) {
+                final VariantContext.Type variantType = typeOfVariant(referenceAllele, altAllele);
+
+                if (variantType == VariantContext.Type.INDEL) {
+                    if (isIndelInThePileupElement(pileupElement, referenceAllele, altAllele)) {
+                        pileupAllele = altAllele;
+                    }
+
+                } else if (variantType == VariantContext.Type.MNP || variantType == VariantContext.Type.SNP) {
+                    if (doesReadContainAllele(pileupElement, altAllele) == ReadContainAllele.TRUE) {
+                        pileupAllele = altAllele;
+                    }
+                }
+            }
+        } else {
+            pileupAllele = referenceAllele;
+        }
+
+        if ((pileupAllele != null) && (getMinBaseQualityForAlleleInRead(pileupElement, pileupAllele) < minBaseQualityCutoff)) {
+            pileupAllele = null;
+        }
+
+        return pileupAllele;
+    }
+
+    private static boolean isIndelInThePileupElement(final PileupElement pileupElement, final Allele referenceAllele, final Allele altAllele) {
+        boolean isAltAlleleInThePileup = false;
+
+        // Check insertion
+        if (pileupElement.isBeforeInsertion()) {
+            final int insertionLength = pileupElement.getLengthOfImmediatelyFollowingIndel();
+            if (insertionLength == pileupElement.getLengthOfImmediatelyFollowingIndel()) {
```

Comment:
Do my eyes deceive me or is this a tautology?

Reply:
Ha!  I'm surprised IntelliJ did not see that.

Done